tests/testthat/test_ExtractData_Soils.R

```diff
@@ -18,23 +19,26 @@ do_skip <- c(
   # mimmics skip_on_appveyor():
   identical(tolower(Sys.getenv("APPVEYOR")), "true"))
 
-suppressWarnings(is_online <-
-                   !inherits(try(close(url(getOption("repos"), open = "r")), silent = TRUE),
-                             "try-error"))
 
-
-if (!any(do_skip) && is_online) {
+if (!any(do_skip)) {
   # =============================================================================
   # Tests designed to test the underlining structures created
   # from soil extraction functions.
   # =============================================================================
   
-  # set stage manually so that future changes won't cause the test to fail ======
   # setup file paths
   fnames_in <- environment()
   fnames_in$fslayers <- file.path("/home/natemccauslin/Desktop/Dryland Ecology/rSFSW2/tests/test_data/TestPrj4/1_Input/SWRuns_InputData_SoilLayers_v9.csv")
   fnames_in$fsoils <- "/home/natemccauslin/Desktop/Dryland Ecology/rSFSW2/tests/test_data/TestPrj4/1_Input/datafiles/SWRuns_InputData_soils_v12.csv"
+  # directory where extertnal soils data is located
   dir_ex_soil <- "/media/natemccauslin/SOILWAT_DATA/GIS/Data/Soils/"
+  
+  # if any of the file paths above are invalid, skip all tests and setup
+  if (!all(file.exists(c(fnames_in$fslayers, fnames_in$fsoils, dir_ex_soil)))) {
+    skip("File paths incorrectly configured for Soils Extraction tests")
```

Comment:
I'm repeating myself `skip` doesn't work outside `test_that` blocks.

Why don't you run the package tests on another machine that doesn't have `GriddedGlobalV5` installed? Something like
```
Sys.setenv(NOT_CRAN = "true")
Sys.setenv(RSFSW2_ALLTESTS = "true")
devtools::test()
```

Reply:
It seems to work, do you mean you don't like it outside of testthat?

Reply:
You are correct `skip` does work correctly outside of `test_that` blocks -- this must be a new feature of more recent `testthat` versions because it didn't work in earlier versions when I checked it out originally several years back (e.g., discussion on this https://github.com/r-lib/testthat/commit/b1e41a045531c443afbdcc62f672747c68aeb754).

Apparently, the devel version of `testthat` also introduced an `skip_if_offline` (https://github.com/r-lib/testthat/commit/be8e6b6ae87642c157c5ed6510076ba37e1bc0ed) which will be nice for us!

Ok, I am fine with your `skip`s, but there is still the problem that the tests are not skipped and fail if the correct external files are not available.